source/Calamari.Common/Features/Processes/Semaphores/SystemSemaphoreManager.cs

```diff
@@ -90,24 +94,33 @@ IDisposable AcquireMutex(string name, string waitMessage)
             }
 
             return new Releaser(() =>
-            {
-                mutex.ReleaseMutex();
-                mutex.Dispose();
-            });
+                                {
+                                    mutex.ReleaseMutex();
+                                    mutex.Dispose();
+                                });
         }
 
-        static Semaphore CreateGlobalSemaphoreAccessibleToEveryone(string name)
+        Semaphore CreateGlobalSemaphoreAccessibleToEveryone(string name)
         {
             var semaphoreSecurity = new SemaphoreSecurity();
             var everyone = new SecurityIdentifier(WellKnownSidType.WorldSid, null);
             var rule = new SemaphoreAccessRule(everyone, SemaphoreRights.FullControl, AccessControlType.Allow);
 
             semaphoreSecurity.AddAccessRule(rule);
 
-            bool createdNew;
+            //we try and create/acquire a global semaphore with some retry
+            var semaphore = semaphoreAcquisitionPipeline.Execute(() => new Semaphore(1,1, name));
```

Comment:
You can move this up to the caller of this method

Reply:
done in [7a92457](https://github.com/OctopusDeploy/Calamari/pull/1414/commits/7a92457fab930a967c63fd150d1d617121dc10f1)